variables.tf

```diff
@@ -65,6 +65,19 @@ variable "enable_victoria_metrics_auth" {
   type        = bool
 }
 
+### Top-level Variables
+
+#######
+### Networking
+#######
+
+variable "service_domain" {
```

Comment:
I'm not sure of a workaround here, but requiring this input variable does introduce some annoyance when we're not using istio (and it's a shame that variable definitions in terraform can only reference themselves in a condition validation, otherwise we could chain this as a requirement when `enable_kiali_operator = true`).

And giving it a `null` or meaningless default could be overlooked and send people down a rabbit hole while they chase something that's broken.

But I still suggest we make this default to `null` in the root module until we can find a solution (which will also need to work for how the `istio-operator` submodule gets composed in the `terraform-*-cloud` modules)

Reply:
I feel you should make peace with the concept of having a composite module for sn-cloud-hosted, that composes the individual modules.  It is not practical to have a shared `sn-bootstrap` and your attempt to make everything optional is not working out.